geomesa-spark/geomesa-spark-sql/src/main/scala/org/apache/spark/sql/SQLSpatialAccessorFunctions.scala

```diff
@@ -0,0 +1,87 @@
+package org.apache.spark.sql
+
+import com.vividsolutions.jts.geom._
+
+object SQLSpatialAccessorFunctions {
+  val ST_Boundary: Geometry => Geometry = geom => geom.getBoundary
+  val ST_CoordDim: Geometry => Int = geom => {
+    val coord = geom.getCoordinate
+    if (coord.z.isNaN) 2 else 3
```

Comment:
The `getCoordinateSequence` method isn't in the Geometry class? Is it possible to mix 2d/3d sub-geometries in a Geometry? Maybe in a GeometryCollection? I could see this function failing if that's the case. It's also unclear if this function is supposed to handle empty geometries gracefully (e.g. return a 0 maybe): http://www.postgis.net/docs/ST_CoordDim.html